Sources/StreamChat/Workers/ChannelListUpdater.swift

```diff
@@ -51,23 +52,34 @@ class ChannelListUpdater: Worker {
                         
                         let localQueryCIDs = Set(queryDTO.channels.compactMap { try? ChannelId(cid: $0.cid) })
                         let remoteQueryCIDs = Set(channelListPayload.channels.map(\.channel.cid))
-                        let queryAlreadySynched = remoteQueryCIDs.intersection(synchedChannelIds)
-                        
-                        // We are going to unlink & clear those channels that are not present in the remote query,
-                        // and that have not been synched nor watched. Those are outdated.
-                        let cidsToRemove = localQueryCIDs
-                            .subtracting(queryAlreadySynched)
-                            .subtracting(watchedChannelIds)
-                        
-                        for cid in cidsToRemove {
+
+                        let updatedChannels = synchedChannelIds.union(watchedChannelIds)
+                        let localNotInRemote = localQueryCIDs.subtracting(remoteQueryCIDs)
+                        let localInRemote = localQueryCIDs.intersection(remoteQueryCIDs)
+
+                        // We unlink those local channels that are no longer in remote
+                        for cid in localNotInRemote {
                             guard let channelDTO = session.channel(cid: cid) else { continue }
-                            
-                            channelDTO.resetEphemeralValues()
-                            channelDTO.messages.removeAll()
                             queryDTO.channels.remove(channelDTO)
                         }
+
+                        // We are going to clean those channels that are present in the both the local and remote query,
+                        // and that have not been synched nor watched. Those are outdated, can contain gaps.
+                        let cidsToClean = localInRemote.subtracting(updatedChannels)
+                        session.cleanChannels(cids: cidsToClean)
```

Comment:
I would only make `RefetchChannelListQueryOperation` operation responsible for unlinking channels from a query and move channels clean-up fully to `CleanUnwantedChannelsOperation` as it makes easier to understand what's happening.

If there's a channel `C1` locally and it is linked to `Q1` and `Q2` local queries, then when the 1st page for `Q1` is fetched `C1` can be outside this page but can still be part of `Q2's` first page. In that case it's easier to make a decision which channels are/can get outdated when all queries are refetched.